inc/sainat.h

```diff
@@ -0,0 +1,653 @@
+     *
+     * @type sai_ip4_t
+     * @flags CREATE_AND_SET
+     * @default 0.0.0.0
+     * @validonly SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_DESTINATION_NAT or SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_DOUBLE_NAT
+     */
+    SAI_NAT_ENTRY_ATTR_DST_IP,
+
+    /**
+     * @brief Mask for destination IPv4 address in packet.
+     *
+     * @type sai_ip4_t
+     * @flags CREATE_AND_SET
+     * @default 0.0.0.0
+     * @validonly SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_DESTINATION_NAT or SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_DOUBLE_NAT
+     */
+    SAI_NAT_ENTRY_ATTR_DST_IP_MASK,
+
+    /**
+     * @brief Replace L4 source port in packet.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @isvlan false
+     * @default 0
+     */
+    SAI_NAT_ENTRY_ATTR_L4_SRC_PORT,
+
+    /**
+     * @brief Replace L4 destination port in packet.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @isvlan false
+     * @default 0
+     */
+    SAI_NAT_ENTRY_ATTR_L4_DST_PORT,
+
+    /**
+     * @brief Enable/disable packet count
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_NAT_ENTRY_ATTR_ENABLE_PACKET_COUNT,
+
+    /**
+     * @brief Per NAT entry packet count
+     *
+     * @type sai_uint64_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_NAT_ENTRY_ATTR_PACKET_COUNT,
+
+    /**
+     * @brief Enable/disable byte count
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_NAT_ENTRY_ATTR_ENABLE_BYTE_COUNT,
+
+    /**
+     * @brief Per NAT entry byte count
+     *
+     * @type sai_uint64_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_NAT_ENTRY_ATTR_BYTE_COUNT,
+
+    /**
+     * @brief NAT entry hit bit clear on read flag
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_NAT_ENTRY_ATTR_HIT_BIT_COR,
+
+    /**
+     * @brief Per NAT entry hit bit state
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_NAT_ENTRY_ATTR_HIT_BIT,
+
+    /**
+     * @brief End of NAT Entry attributes
+     */
+    SAI_NAT_ENTRY_ATTR_END,
+
+    /** Custom range base value */
+    SAI_NAT_ENTRY_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_NAT_ENTRY_ATTR_CUSTOM_RANGE_END
+
+} sai_nat_entry_attr_t;
+
+/**
+ * @brief NAT entry keys
+ */
+typedef struct _sai_nat_entry_key_t
+{
+
+    /**
+     * @brief IPv4 source address
+     */
+    sai_ip4_t src_ip;
+
+    /**
+     * @brief IPv4 destination address
+     */
+    sai_ip4_t dst_ip;
+
+    /**
+     * @brief IP protocol value
+     */
+    sai_uint8_t proto;
+
+    /**
+     * @brief IP layer 4 source port
+     */
+    sai_uint16_t l4_src_port;
+
+    /**
+     * @brief IP layer 4 destination port
+     */
+    sai_uint16_t l4_dst_port;
+
+} sai_nat_entry_key_t;
```

Comment:
This entry is a 5-tuple. SNAT and DNAT use 3-tuples and there is no validation here.

Reply:
condition check has been added. Please look at the latest updates.
  /**
     * @brief Mask for source IPv4 address in packet.
     *
     * @type sai_ip4_t
     * @flags CREATE_AND_SET
     * @default 0.0.0.0
     * @validonly SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_SOURCE_NAT or SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_DOUBLE_NAT
     */
    SAI_NAT_ENTRY_ATTR_SRC_IP_MASK,